### ISSUE TITLE
Bot occasionally crashes and dies.

### ISSUE THREAD

**Issue by neurological1** [kolton/d2bot-with-kolbot#737]
I'm trying to bot a sorc solo from 98 to 99, and it'll do great for awhile, but after i get like 10-15% EXP into 98 I keep incurring on this scene:

https://i.gyazo.com/ade580e2e767100c4f2216e7b4ae01e2.png

It always gives that error message and then the area, whether at Nihlathak or Throne of Destruction etc, then when it logs back in my char has died. Does this mean lag is causing this (because of the CD key in use?)

Thanks.

**Comment by DarkHorseDre** [comment]
no, cdkey in use wont cause lag - bot wont start if key in use.

you need to watch the bot to see how and why it dies. nith and baal area vipers etc can kill a soso depending on scenario. the bot doesn't fight as well as we would, so you need to watch to see what happens.  settings such as config.telestomp and dodge can make a difference.  the skill used on nith and natures peace are other things - just watch the bot

**Comment by neurological1** [comment]
> no, cdkey in use wont cause lag - bot wont start if key in use.
> 
> you need to watch the bot to see how and why it dies. nith and baal area vipers etc can kill a soso depending on scenario. the bot doesn't fight as well as we would, so you need to watch to see what happens. settings such as config.telestomp and dodge can make a difference. the skill used on nith and natures peace are other things - just watch the bot

The thing is the sorc does fine, but his death always seems to be accompanied preemptively by a crash message. It seems to crash, tries to log back in, CD key in use, waits, logs back in finally, goes into a game, and it sees the lesser XP and registers it as a death... but I don't know whats crashing it. 

Is there a log file anywhere where it might show my ping at the time or anything?

**Comment by DarkHorseDre** [comment]
ok so always crash then death.. sounds like client/server lag?

d2bs.log is one log - it references crash logs in the d2 install `BlizzardError` folder, but dont think that will help.
I don't see any files with my crash info (Aside from console.rtf which is the same info you see in the console).

re: ping - the console and the console.rtf show the ping, heres a random extract from my rtf:
`Life Chicken (0/1124) to Devilkin in Tower Cellar Level 5. Ping: 16`

**Comment by alfton** [comment]
I can concure this, bot does crash when lag occour. Out of 50 run, you will have arround 15-17 crashes. I did test this bot on a tcp/ip game on local net, there never occour crashes there, only some of the script complainin about not finding som bosses.
